### PR TITLE
LieAlgebras: Enhance module interface

### PR DESCRIPTION
**Pull request by lgoettgens**
Currently, there is some weird behavior when two identical modules have been constructed in different ways, e.g. some module and its 1st exterior power.
```julia
julia> L = special_orthogonal_lie_algebra(QQ, 4)
LinearLieAlgebra (⊆ gl_4) over Rational Field

julia> std_V = standard_module(L)
StdModule of LinearLieAlgebra (⊆ gl_4) over Rational Field

julia> V = exterior_power(std_V, 1)
<StackOverflowError>

julia> is_exterior_power(std_V)
true
```
This PR first adds tests modeling the intended behavior and then approaches to fix the above accordingly.

While on my way, I may add some more module constructions.

### CODE REVIEW

File: experimental/LieAlgebras/src/LieAlgebraModule.jl [hunk 1]
```diff
@@ -273,16 +292,28 @@ end
 #
 ###############################################################################
 
+function Base.:(==)(V1::LieAlgebraModule{C}, V2::LieAlgebraModule{C}) where {C<:RingElement}
+  return V1.L == V2.L &&
+         V1.dim == V2.dim &&
```
Comment (fingolfin) [inline experimental/LieAlgebras/src/LieAlgebraModule.jl]: The comparison should be ordered by cost, first the cheap comparisions (dim, s), then the more expensive ones

File: experimental/LieAlgebras/src/LieAlgebraModule.jl [hunk 2]
```diff
@@ -337,6 +376,16 @@ function is_tensor_power(V::LieAlgebraModule{C}) where {C<:RingElement}
   return get_attribute(V, :type, :fallback) == :tensor_power
 end
 
+function base_module(V::LieAlgebraModule{C}) where {C<:RingElement}
+  @req is_dual(V) || is_exterior_power(V) || is_symmetric_power(V) || is_tensor_power(V) "Not a power module."
+  return get_attribute(V, :base_module)
```
Comment (fingolfin) [inline experimental/LieAlgebras/src/LieAlgebraModule.jl]: Ideally (if you can( add a type assertion to improve type stability
```suggestion
  return get_attribute(V, :base_module)::THE_TYPE_OF_THE_BASE_MODULE
```

Comment (lgoettgens) [reply experimental/LieAlgebras/src/LieAlgebraModule.jl]: At this place, that is easy to do.

File: experimental/LieAlgebras/src/LieAlgebraModule.jl [hunk 3]
```diff
@@ -337,6 +376,16 @@ function is_tensor_power(V::LieAlgebraModule{C}) where {C<:RingElement}
...
 end
 
+function base_module(V::LieAlgebraModule{C}) where {C<:RingElement}
+  @req is_dual(V) || is_exterior_power(V) || is_symmetric_power(V) || is_tensor_power(V) "Not a power module."
+  return get_attribute(V, :base_module)::LieAlgebraModule{C}
+end
+
+function base_modules(V::LieAlgebraModule{C}) where {C<:RingElement}
+  @req is_direct_sum(V) "Not a direct sum module."
+  return get_attribute(V, :base_modules)::Vector{LieAlgebraModule{C}}
```
Comment (lgoettgens) [inline experimental/LieAlgebras/src/LieAlgebraModule.jl:403]: Here, I changed it from an abstract `Tuple{Vararg{LieAlgebraModule{C}}}` to a concrete `Vector{LieAlgebraModule{C}}`. @fingolfin

**Comment by lgoettgens** [comment]
This is ready for review and merge now. Since I expect some small conflict with #2207, I would rather merge this one first and deal with #2207 later. I anticipate a discussion or at least some ugly hacks there.

**Comment by lgoettgens** [comment]
This is ready for review and merge now. Since I expect some small conflict with https://github.com/oscar-system/Oscar.jl/pull/2207, I would rather merge this one first and deal with https://github.com/oscar-system/Oscar.jl/pull/2207 later. I anticipate a discussion or at least some ugly hacks there.

**Review by fingolfin — APPROVED** [review]
Fine by me

**Pull request merged**